at_radial_gauges/lib/radial_gauges/scale_gauge.dart

```diff
@@ -0,0 +1,322 @@
+        maxValue: widget.maxValue,
+        minValue: widget.minValue,
+        maxDegrees: 300);
+
+    double upperBound = Utils.degreesToRadians(300);
+
+    animationController = AnimationController(
+        duration: Utils.getDuration(
+            isAnimate: widget.isAnimate, userMilliseconds: widget.duration),
+        vsync: this,
+        upperBound: upperBound);
+
+    animation = Tween<double>().animate(animationController)
+      ..addListener(() {
+        if (animationController.value == sweepAngleRadian) {
+          animationController.stop();
+        }
+
+        setState(() {});
+      });
+
+    animationController.forward();
+  }
+
+  @override
+  void dispose() {
+    animationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (animationController.value !=
+        Utils.actualValueToSweepAngleRadian(
+            actualValue: widget.actualValue,
+            maxValue: widget.maxValue,
+            minValue: widget.minValue,
+            maxDegrees: 300)) {
+      animationController.animateTo(
+          Utils.actualValueToSweepAngleRadian(
+              actualValue: widget.actualValue,
+              maxValue: widget.maxValue,
+              minValue: widget.minValue,
+              maxDegrees: 300),
+          duration: Utils.getDuration(
+              isAnimate: widget.isAnimate, userMilliseconds: widget.duration));
+    }
+
+    return FittedBox(
+      child: SizedBox(
+        child: Column(
+          children: [
+            widget.titlePosition == TitlePosition.top
+                ? SizedBox(
+                    height: 20,
+                    child: widget.title,
+                  )
+                : const SizedBox(
+                    height: 20,
+                  ),
+            SizedBox(
+              height: widget.size,
+              width: widget.size,
+              child: Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: CustomPaint(
+                  painter: RangePointerGaugePainter(
+                    sweepAngle: animationController.value,
+                    pointerColor: widget.arcColor,
+                    needleColor: widget.needleColor,
+                    minValue: widget.minValue,
+                    maxValue: widget.maxValue,
+                    actualValue: widget.actualValue,
+                    decimalPlaces: widget.decimalPlaces,
+                  ),
+                ),
+              ),
+            ),
+            const SizedBox(
+              height: 10,
+            ),
+            widget.titlePosition == TitlePosition.bottom
+                ? SizedBox(
+                    height: 30,
+                    child: widget.title,
+                  )
+                : const SizedBox(
+                    height: 20,
+                  )
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class RangePointerGaugePainter extends CustomPainter {
```

Comment:
Can we keep `RangePointerGaugePainter` in a separate file?

Reply:
Actually the more appropriate name should be ScaleGaugePainter, I'll push the change shortly. I kept it in the same file since that class is only being used by the ScaleGauge stateful widget. 

Given the above explanation what do you recommend we do?